doc/articles/contributing/guidelines/creating-tests.md

```diff
@@ -0,0 +1,69 @@
+
+UI tests can mimic the actions of a user:
+
+ - tapping buttons and other UI elements
+ - entering keyboard input
+ - scrolling, swiping, and other gestures
+
+ They can verify assertions about the state of the app:
+
+  - Text labels
+  - any DependencyProperty value
+  - onscreen bounds of a view element
+  - comparing screenshots at different stages of the app, and asserting equality or inequality
+
+A complete set of instructions for authoring UI tests is available [here](../../uno-development/working-with-the-samples-apps.md).
+
+### .NET Framework unit tests (`Uno.UI.Tests`)
+
+These are 'classic' unit tests which are built against a mocked version of the `Uno.UI` assembly targetting .NET Framework. 
+
+These tests are ideal for testing platform-agnostic parts of the code, such as the dependency property system, or panel measurement logic. They [have access to](https://github.com/unoplatform/uno/blob/af5365331d87a80ed4dd83b9e4839cc0d4a0ee5b/src/Uno.UI/AssemblyInfo.cs#L3) internal Uno.UI members, and the mocking layer allows values to be read or written which can't be accessed via the 'real' API.
+
+### Platform runtime unit tests (`Uno.UI.RuntimeTests`)
+
+Again these are 'classic' unit tests, but they are run 'in-process' on the actual target platform, using the 'real' Uno.UI assemblies. They can be run locally through the [Unit Tests Runner](https://github.com/unoplatform/uno/blob/master/src/SamplesApp/SamplesApp.Shared/Samples/UnitTests/UnitTestsPage.xaml) sample in the SamplesApp.
+
+These tests are useful for testing behavior which can run synchronously and where correctness can be asserted programmatically. Relative to the .NET Framework tests, they have the advantage that platform-dependent behavior can be tested, but the disadvantage of being restricted to the public API surface.
+
+### Xaml code generation tests (`XamlGenerationTests`)
+
+These specifically target [the parser](https://github.com/unoplatform/uno/tree/master/src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator) which generates C# code from Xaml files. They are 'tests' in a simple sense that if the parser throws an error, or if it generates invalid C# code, they will fail the CI build. 
+
+If you want to actually test that generated Xaml produces correct behavior, which will be the case most of the time, you should use one of the other test types.
+
+## Which type of test should I create?
+
+The UI tests are in some sense the most powerful means of testing behavior, since they can simulate real user input and verify the state of the running app. However they're also typically more time-consuming to author, and take longer to run several orders of magnitude, which affects both your own personal development loop and also the running time of the CI build. Sometimes a UI test is a sledgehammer where you only need... some... smaller hammer. 
```

Comment:
```suggestion
The UI tests are in some sense the most powerful means of testing behavior, since they can simulate real user input and verify the state of the running app. However they're also typically more time-consuming to author, and take several orders of magnitude longer to run , which affects both your own personal development loop and also very importantly the running time of the CI build. Sometimes a UI test is a sledgehammer where you only need... some... smaller hammer. 
```